### DISCUSSION TITLE
Is vm.broadcast() and vm.broadcast a MUST to deploy a contract ?

### DISCUSSION THREAD

**Discussion by GhislainTselone** [Cyfrin/foundry-full-course-cu#1253] [General]
Observation: I noticed that there are some contract that have been deployed between :

```solidity
vm.startBroadcast()
...... 
vm.stopBroadcast()
Questions:
```

1.Why some need it and why others don't ?
2.If i try to deploy a contract without it, it won't work ?

**Comment by ghost** [comment]
These cheatcodes are used by Foundry to collect the transactions in scripts that you want to send to the blockchain.

I recommend you to read: https://book.getfoundry.sh/tutorials/solidity-scripting
